src/components/QuizGame/index.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from "react";
+import { fetchQuizQuestions } from "../../components/QuizGameApi";
+import QuestionCard from "../../components/QuizGameQuestions";
+import "./index.css";
+
+const DIFFICULTY = {
+  EASY: "easy",
+  MEDIUM: "medium",
+  HARD: "hard",
+};
+const TOTAL_QUESTIONS = 10;
+
+const QuizGame = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [questions, setQuestions] = useState([]);
+  const [currentQuestionNumber, setCurrentQuestionNumber] = useState(0);
+  const [userAnswers, setUserAnswers] = useState([]);
+  const [score, setScore] = useState(0);
+  const [isGameStart, setIsGameStart] = useState(false);
+
+  const startQuiz = async () => {
+    setIsLoading(true);
+    setIsGameStart(true);
+    const newQuestions = await fetchQuizQuestions(
+      TOTAL_QUESTIONS,
+      DIFFICULTY.EASY
+    );
+    setQuestions(newQuestions);
+    setScore(0);
+    setUserAnswers([]);
+    setCurrentQuestionNumber(0);
+    setIsLoading(false);
+  };
+  const checkAnswer = (e) => {
+    if (isGameStart) {
+      const userChosenAnswer = e.currentTarget.value;
+      const checkCorrectAnswer =
+        questions[currentQuestionNumber].correct_answer;
+      const isCorrect = checkCorrectAnswer === userChosenAnswer;
+      if (isCorrect) {
+        setScore((prev) => prev + 1);
+        const answerObject = {
+          question: questions[currentQuestionNumber].question,
+          userChosenAnswer,
+          isCorrect,
```

Comment:
so answerObject is obviously and object but when you add the variables `useChosenAnswer` and `isCorrect` to it as items, what would be the key to those items? will it look something like this:
`{...,
userChosenAnswer: (user's answer),
isCorrect: (isCorrect's value)}
`
I realized that you use this kind of syntax in your code and i'm not familiar with it, so i'm just writing this to understand what's going on

Reply:
it is declared above in the line 37, but i still added it as a keys so it will be clearer, i am repeating my self in this case lets check if louis ok with it. @louisrli